src/Spago/Git.purs

```diff
@@ -40,14 +41,19 @@ runGit args cwd = ExceptT do
     Right r -> Right r.stdout
     Left r -> Left r.stderr
 
-fetchRepo :: forall a b. { git :: String, ref :: String | a } -> FilePath -> Spago (GitEnv b) (Either (Array String) Unit)
+determineShouldCache :: forall b. String -> FilePath -> Spago (GitEnv b) Boolean
+determineShouldCache ref path = do
+  showRef <- Except.runExceptT $ runGit_ [ "show-ref", "--verify", "refs/heads/" <> ref ] (Just path)
+  pure $ isLeft showRef
+
+fetchRepo :: forall a b. { git :: String, ref :: String | a } -> FilePath -> Spago (GitEnv b) (Either (Array String) Boolean)
 fetchRepo { git, ref } path = do
   repoExists <- FS.exists path
   { offline } <- ask
   case offline, repoExists of
     Offline, true -> do
       logDebug $ "Found " <> git <> " locally, skipping fetch because we are offline"
-      pure $ Right unit
+      Right <$> determineShouldCache ref path
```

Comment:
`fetchRepo` is used throughout the whole project and not only when fetching dependencies. We shouldn't run this check here if possible